apps/site/components/Common/Search/index.tsx

```diff
@@ -1,61 +1,132 @@
+      siteSection.toLowerCase() === 'docs' ? `/${path}` : `/${locale}/${path}`,
+    section: 'siteSection',
+  };
 
   return (
     <>
-      <button
-        type="button"
-        onClick={openSearchBox}
-        className={styles.searchButton}
+      <OramaSearchButton
+        style={{ flexGrow: 1 }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
         aria-label={t('components.search.searchBox.placeholder')}
       >
-        <MagnifyingGlassIcon className={styles.magnifyingGlassIcon} />
-
         {t('components.search.searchBox.placeholder')}
-        <kbd
-          title={`${osCommandKey} K`}
-          className={classNames(styles.shortcutIndicator, {
-            'opacity-0': isOSLoading,
-          })}
-        >
-          <abbr>{osCommandKey} K</abbr>
-        </kbd>
-      </button>
-
-      {isOpen ? <WithSearchBox onClose={closeSearchBox} /> : null}
+      </OramaSearchButton>
+
+      <OramaSearchBox
+        index={{ api_key: ORAMA_CLOUD_API_KEY, endpoint: ORAMA_CLOUD_ENDPOINT }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
+        sourceBaseUrl={BASE_URL}
+        sourcesMap={sourceMap}
+        resultMap={resultMap}
+        facetProperty="siteSection"
+        linksTarget="_self"
+        highlightTitle={{
+          caseSensitive: false,
+          HTMLTag: 'b',
+          CSSClass: 'font-bold',
+        }}
+        searchParams={DEFAULT_ORAMA_QUERY_PARAMS}
+        suggestions={DEFAULT_ORAMA_SUGGESTIONS}
+        chatMarkdownLinkHref={({ href }) => {
+          if (!href) return href;
+
+          const currentPageHost = `${window.location.protocol}//${window.location.host}`;
+          const docsBaseUrl = `${currentPageHost}/docs`;
+          const lowerCaseHref = href.toLowerCase();
+
+          // Keep original link if URl should not have locale
+          if (
+            !lowerCaseHref.startsWith(currentPageHost) ||
+            lowerCaseHref.startsWith(docsBaseUrl)
+          ) {
+            return href;
+          }
+
+          const before = lowerCaseHref.slice(0, currentPageHost.length);
+          const after = lowerCaseHref.slice(currentPageHost.length);
+          return `${before}/${locale}${after}`;
+        }}
+        onAnswerSourceClick={event => {
+          event.preventDefault();
+
+          const currentPageHost = `${window.location.protocol}//${window.location.host}`;
+          const { path } = event.detail.source;
+
+          if (!path.startsWith('docs/')) {
+            window.open(`${currentPageHost}/${locale}/${path}`, '_blank');
+          } else {
+            window.open(`${currentPageHost}/${path}`, '_blank');
+          }
+        }}
+        onSearchResultClick={event => {
+          event.preventDefault();
+
+          const currentPageHost = `${window.location.protocol}//${window.location.host}`;
+
+          router.push(`${currentPageHost}/${event.detail.result.path}`, {
+            locale: undefined,
```

Comment:
Explain why this is set to undefined within inline docs

Reply:
The URL provided on the callback event (event.details.result.path) always contains the locale because of the resultMap property. URLs are already good to be clicked  on searchbox, with the locale being dynamically appended.

In summary, router.push from @navigator always append the locale for not undefined locale. It is bad because:
A) For /docs paths, we can not have the locale
B) We already add the locale on resultMap

Reply:
I think you misunderstood me. Im asking you to add inline commentary on the code of why you passed this undefined, other people might not understand the why,